src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Configuration.targets

```diff
@@ -28,30 +28,44 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <!--
-      Configure MVC specific configuration if the application uses a configuration that supports MVC.
+      Resolve MVC specific configuration depending on the presence of MVC specific Razor files.
   -->
-  <PropertyGroup Condition="'$(AddRazorSupportForMvc)' == 'true'">
-    <!--
-      MVC uses a ProvideApplicationPartFactoryAttribute on the generated assembly to load compiled views from assembly. Set this to false, to prevent generating this attribute.
-    -->
-    <GenerateProvideApplicationPartFactoryAttribute Condition="'$(GenerateProvideApplicationPartFactoryAttribute)'==''">true</GenerateProvideApplicationPartFactoryAttribute>
+  <Target Name="_ResolveMvcSpecificRazorConfiguration"
+    DependsOnTargets="ResolveRazorGenerateInputs"
+    Condition="'$(Language)' == 'C#' AND '$(AddRazorSupportForMvc)' == 'true'">
 
-    <!--
-      Determines if the Sdk is allowed to add additional attributes to the project assembly.
-      For instance, MVC will generally want to add attributes to support view discovery and runtime compilation.
-    -->
-    <GenerateRazorAssemblyInfo Condition="'$(GenerateRazorAssemblyInfo)'==''">true</GenerateRazorAssemblyInfo>
+    <Message Text="@(RazorGenerateWithTargetPath)" Importance="High" />
 
-    <!--
-      MVC will generally want to add support for runtime compilation, but only for applications.
-    -->
-    <GenerateRazorHostingAssemblyInfo Condition="'$(GenerateRazorHostingAssemblyInfo)'=='' AND '$(OutputType)'=='exe'">$(GenerateRazorAssemblyInfo)</GenerateRazorHostingAssemblyInfo>
+   <PropertyGroup Condition="'@(RazorGenerate->Count())' != '0'">
```

Comment:
MSBuild noob question, it's never impossible here for RazorGenerate to be an undefined equivalent correct?

Reply:
Undefined collections are the equivalent of `Enumerable.Empty`. Similarly, undefined properties evaluate to "empty"